### PR TITLE
Track resize feature

### PR DESCRIPTION
**Pull request by nicoabie**
Pull request for issue #2539

### CODE REVIEW

**Comment by lucaswoj** [comment]
I'm happy to 🚢  this once you add a unit test! Thanks @nicoabie.

File: js/util/browser.js [hunk 1]
```diff
@@ -9,6 +9,13 @@
  * it's used for tests.
  */
 
+var window = {
+    addEventListener: function() {},
+    removeEventListener: function() {}
+};
+
+exports.window = window;
+
```
Comment (mourner) [inline js/util/browser.js]: also needs to be added to `browser/browser.js` so that it doesn't break the non-test build

Comment (nicoabie) [inline js/util/browser.js]: Gotcha @mourner, I'll add an export for window as browser/browser.js assumes there is already a global 'window' object defined.
Thanks for the hint.

**Comment by nicoabie** [comment]
@lucaswoj sorry for the many commits but I'm having issues with gl dependecy as described in #1891 so have to commit blindly and check for errors using the ci.

**Comment by lucaswoj** [comment]
Thanks @nicoabie! Tweaked, squashed, and merged as cb814378a30c634dce942dcfbd31f15d8e538b2d!

**Pull request closed without merging**